cmd/test_stores/main.go

```diff
@@ -261,26 +326,35 @@ func (t *traceGenerator) generateTrace(out chan *centralstore.CentralSpan, stop
 		case <-stop:
 			return
 		case <-ticker.C:
+			_, span := t.tracer.Start(ctx, "send_span")
+			addSpanFields(span, map[string]interface{}{
+				"trace_id": traceid,
+				"span_id":  trace.Spans[counter-1].SpanID,
+				"index":    counter - 1,
+			})
 			// only send the span if it's ours
 			if t.mySpan(traceid, trace.Spans[counter-1].SpanID) {
-				fmt.Printf(" sending %d tid: %s spid: %s\n", counter-1, trace.TraceID, trace.Spans[counter-1].SpanID)
+				addSpanField(span, "sent", true)
 				out <- trace.Spans[counter-1]
 			} else {
-				fmt.Printf("skipping %d tid: %s spid: %s\n", counter-1, trace.TraceID, trace.Spans[counter-1].SpanID)
+				addSpanField(span, "sent", false)
 			}
+			span.End()
```

Comment:
Normally better to `defer span.End()` 😄

Reply:
I thought the defer would trigger as part of the case statement evaluation?

Reply:
[Defers are only called in reverse order after the function containing them returns.](https://go.dev/blog/defer-panic-and-recover)